selftest/uprobe-multi/main.go

```diff
@@ -0,0 +1,238 @@
+package main
+
+import "C"
+import (
+	"bytes"
+	"debug/elf"
+	"encoding/binary"
+	"fmt"
+	"hash/fnv"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	bpf "github.com/aquasecurity/libbpfgo"
+	"github.com/aquasecurity/libbpfgo/helpers"
+)
+
+type Event struct {
+	Cookie uint64
+}
+
+type FunctionInfo struct {
+	Name   string
+	Offset uint64
+}
+
+const (
+	attachBatchSize  = 256
+	bpfProgramName   = "uprobe__test_functions"
+	bpfProgramObject = "main.bpf.o"
+)
+
+var (
+	excludePatterns = []string{"runtime.", "go:", "internal"}
+)
+
+func main() {
+	if len(os.Args) < 3 {
+		fmt.Fprintln(os.Stderr, "wrong syntax")
+		os.Exit(-1)
+	}
+
+	// Executable and expected symbols to be traced as positional arguments.
+	binaryPath := os.Args[1]
+	expectedSymbolNames := strings.Split(os.Args[2], ",")
+
+	// We try to attach uprobes to the maximum amount of functions supported
+	// as possible.
+	symbols, err := getFunSyms(binaryPath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to get function symbols: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Hashmap to correlate a cookie got from BPF to a function.
+	cookieToFunctionInfo := make(map[uint64]FunctionInfo, 0)
```

Comment:
```suggestion
	cookieToFunctionInfo := make(map[uint64]FunctionInfo)
```

The hint in this case can be omitted.